metrics-operator/controllers/common/providers/prometheus/prometheus.go

```diff
@@ -67,3 +67,45 @@ func (r *KeptnPrometheusProvider) EvaluateQuery(ctx context.Context, metric metr
 	}
 	return value, b, nil
 }
+
+func EvaluateQueryWithRange(ctx context.Context, metric metricsapi.KeptnMetric, r *KeptnPrometheusProvider, api prometheus.API) (model.Value, prometheus.Warnings, error) {
+	queryTime := time.Now().UTC()
+	// Get the duration
+	queryInterval, err := time.ParseDuration(metric.Spec.Range.Interval)
+	if err != nil {
+		return nil, nil, err
+	}
+	// Convert type Duration to type Time
+	startTime := queryTime.Add(-queryInterval).UTC()
+	r.Log.Info("Running query: /api/v1/query_range?query=" + metric.Spec.Query + "&start=" + startTime.String() + "&end=" + queryTime.String())
+
+	queryRange := prometheus.Range{
```

Comment:
it seems like the `step` parameter must also be passed when using time range queries (https://prometheus.io/docs/prometheus/latest/querying/api/#range-queries).

This might also be the reason for why the integration tests are failing, as I've found this in the logs:

```
2023-06-22T23:32:11Z	ERROR	KeptnMetric Controller	Failed to evaluate the query	{"Response from provider was:": "", "error": "bad_data: invalid parameter \"step\": zero or negative query resolution step widths are not accepted. Try a positive integer"}
```

Reply:
> What do you suggest keeping the `step` param as? I was trying to keep it as the `fetchIntervalSeconds`

Not quite sure yet what the best value to user here tbh... Can you maybe try to get the response payloads for a range queries with different values for `step` and post them here? E.g. a query with a time range of `5min` and the step set to `fetchIntervalSeconds` and then to `5min`, i.e. the same as the time frame. Maybe that will give us further info what the best approach would be

Reply:
I have used the `step` param as the `range.interval` as if we are to set it to `fetchIntervalSeconds`, we will get multiple values in our result and it would be later difficult for us to choose which value to display in the result field, thus I thought, we can modify this logic once we work on the aggregation functions later.